src/ncp/ncp_base.cpp

```diff
@@ -2486,37 +2488,60 @@ void NcpBase::InsertPropertyHandler_IPV6_ADDRESS_TABLE(uint8_t header, spinel_pr
 {
     spinel_ssize_t parsedLength;
     ThreadError errorCode = kThreadError_None;
-    otNetifAddress netif_addr = {};
+    otNetifAddress *netif_addr = NULL;
     otIp6Address *addr_ptr;
+    uint32_t preferred_lifetime;
+    uint32_t valid_lifetime;
+    uint8_t  prefix_len;
 
     parsedLength = spinel_datatype_unpack(
                        value_ptr,
                        value_len,
                        "6CLL",
                        &addr_ptr,
-                       &netif_addr.mPrefixLength,
-                       &netif_addr.mPreferredLifetime,
-                       &netif_addr.mValidLifetime
+                       &prefix_len,
+                       &preferred_lifetime,
+                       &valid_lifetime
                    );
 
     if (parsedLength > 0)
     {
-        netif_addr.mAddress = *addr_ptr;
-        errorCode = otAddUnicastAddress(&netif_addr);
+        for (unsigned i = 0; i < sizeof(mNetifAddresses) / sizeof(mNetifAddresses[0]); i++)
+        {
+            if (mNetifAddresses[i].mNext == NULL)
```

Comment:
Couple comments:
1. I don't think checking `mNext` is robust (e.g. in the case that the entry happens to be at the tail).  The most reliable way with the current API is to use `otGetUnicastAddresses` and walk the list.
2. Think about handling the case of adding an address that has been added before.  Does that return an error?  Does that simply update `mPrefixLength`, `mPreferredLifetime`, and/or `mValidLifetime`?